### PR TITLE
Rcal 740 Add Data Products documentation to RTD

### LINKED ISSUE #1061: Add documentation for data product types to rtd
_Issue [RCAL-750](https://jira.stsci.edu/browse/RCAL-750) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

Add section to describe Roman data products similar to

<https://jwst-pipeline.readthedocs.io/en/latest/jwst/data_products/product_types.html>

### PR DESCRIPTION
**Pull request by ddavis-stsci**
Resolves [RCAL-740](https://jira.stsci.edu/browse/RCAL-740)

Closes #1061

This PR adds documentation for the science products in the exposure level pipeline. 

**Checklist**
- [N/A] added entry in `CHANGES.rst` under the corresponding subsection
- [N/A] updated relevant tests
- [x] updated relevant documentation
- [x] updated relevant milestone(s)
- [x] added relevant label(s)
- [N/A] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -12,6 +12,24 @@ The Roman Space Telescope Calibration Pipeline
    :align: center
    :alt: Nancy Roman Space Telescope
 
+Welcome to the documentation for the Roman calibration software,
+`romancal`. This package contains the Python software suite for the
+Roman Space Telescope (RST) calibration pipeline, which processes data
+from all Roman instruments by applying various corrections to produce
```
Comment (nden) [inline docs/index.rst:18]: Only WFI, not all instruments

File: docs/index.rst [hunk 2]
```diff
@@ -12,6 +12,24 @@ The Roman Space Telescope Calibration Pipeline
    :align: center
    :alt: Nancy Roman Space Telescope
 
+Welcome to the documentation for the Roman calibration software,
+`romancal`. This package contains the Python software suite for the
```
Comment (nden) [inline docs/index.rst:16]: `romancal` is written as a link but is not actually linked.

File: docs/index.rst [hunk 3]
```diff
@@ -12,6 +12,24 @@ The Roman Space Telescope Calibration Pipeline
...
+Roman Space Telescope (RST) calibration pipeline, which processes data
+from all Roman instruments by applying various corrections to produce
+science-ready, calibrated output products including fully calibrated
+individual exposures as well as high-level data products (mosaics,
+catalogs, etc.). The tools in this package allow users to run and
+configure the pipeline to custom process their Roman data.
+Additionally, the `romancal` package contains the interface to
```
Comment (nden) [inline docs/index.rst:23]: Again `romancal` is in back ticks, which attempt to create a link but the link is not provided.

File: CHANGES.rst [hunk 4]
```diff
@@ -17,8 +17,6 @@ jump detection
 documentation
 -------------
 
-- Update jump step docs [#1035]
```
Comment (nden) [inline CHANGES.rst:20]: Was this change intended?

**Review by nden — CHANGES_REQUESTED** [review]
I left some inline comments.
In addition the Exposure level products say the data is a 4D array. This needs to be corrected.

File: CHANGES.rst (hunk 4 above)
Comment (ddavis-stsci) [reply CHANGES.rst:20]: This went in Dec 7, I think I just moved the text. 
We used to keep the change log in alphabetical order. I think I'll just give up.

File: docs/index.rst (hunk 2 above)
Comment (ddavis-stsci) [reply docs/index.rst]: Fixed the link.

File: docs/index.rst (hunk 1 above)
Comment (ddavis-stsci) [reply docs/index.rst]: Fixed.

File: docs/index.rst (hunk 3 above)
Comment (ddavis-stsci) [reply docs/index.rst]: Fixed.

File: docs/roman/data_products/science_products.rst [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+Exposure raw data products are designated by a file name
+suffix of "uncal." These files usually contain only the raw detector pixel values
+from an exposure, with the addition of meta data associated with the exposure. The
+resultantdq array is an optional array used to flag missing data in the data Formatting
+process.
+
++--------------+----------+------------+-------+-------------------------------+
+| data array   |          | Data Type  | Units | Dimensions                    |
++==============+==========+============+=======+===============================+
+|  data        | Required | uint16     | DN    |  ncols x nrows x nresultants  |
```
Comment (nden) [inline docs/roman/data_products/science_products.rst:19]: Since this is using numpy arrays directly in Python, as opposed through FITS, should the dimensions be listed as 
`nresultants x nrows x ncols`?

Comment (nden) [reply docs/roman/data_products/science_products.rst:19]: If this is the right way to present the dimensions then they need to change in every row in this table.

Comment (ddavis-stsci) [reply docs/roman/data_products/science_products.rst]: Updated

**Review by nden — APPROVED** [review]
Looks good. Thanks!
